packages/swapper/src/swappers/zrx/utils/helpers/helpers.test.ts

```diff
@@ -0,0 +1,143 @@
+      const result = normalizeAmount('586084736227728377283728272309128120398')
+      expect(result).toEqual('586084736227728400000000000000000000000')
+    })
+  })
+
+  describe('getAllowanceRequired', () => {
+    it('should return 0 if the sellAsset symbol is ETH', async () => {
+      const quote = {
+        ...quoteInput,
+        sellAsset: { ...sellAsset, symbol: 'ETH' }
+      }
+      expect(await getAllowanceRequired({ quote, web3: web3Instance, erc20AllowanceAbi })).toEqual(
+        new BigNumber(0)
+      )
+    })
+
+    it('should return sellAmount if allowanceOnChain is 0', async () => {
+      const allowanceOnChain = '0'
+      ;(web3Instance.eth.Contract as jest.Mock<unknown>).mockImplementation(() => ({
+        methods: {
+          allowance: jest.fn(() => ({
+            call: jest.fn(() => allowanceOnChain)
+          }))
+        }
+      }))
+
+      expect(
+        await getAllowanceRequired({ quote: quoteInput, web3: web3Instance, erc20AllowanceAbi })
+      ).toEqual(new BigNumber(quoteInput.sellAmount))
+    })
+
+    it('should thow error if allowanceOnChain is undefined', async () => {
+      const allowanceOnChain = undefined
+      ;(web3Instance.eth.Contract as jest.Mock<unknown>).mockImplementation(() => ({
+        methods: {
+          allowance: jest.fn(() => ({
+            call: jest.fn(() => allowanceOnChain)
+          }))
+        }
+      }))
+
+      await expect(
+        getAllowanceRequired({ quote: quoteInput, web3: web3Instance, erc20AllowanceAbi })
+      ).rejects.toThrow(
+        `No allowance data for ${quoteInput.allowanceContract} to ${quoteInput.receiveAddress}`
+      )
+    })
+
+    it('should return 0 if sellAmount minus allowanceOnChain is negative', async () => {
+      const sellAmount = '100'
+      const allowanceOnChain = '1000'
+      const quote = {
+        ...quoteInput,
+        sellAmount
+      }
+      ;(web3Instance.eth.Contract as jest.Mock<unknown>).mockImplementation(() => ({
+        methods: {
+          allowance: jest.fn(() => ({
+            call: jest.fn(() => allowanceOnChain)
+          }))
+        }
+      }))
+
+      expect(await getAllowanceRequired({ quote, web3: web3Instance, erc20AllowanceAbi })).toEqual(
+        new BigNumber(0)
+      )
+    })
+
+    it('should return sellAsset minus allowanceOnChain', async () => {
+      const sellAmount = '1000'
+      const allowanceOnChain = '100'
+      const quote = {
+        ...quoteInput,
+        sellAmount
+      }
+      ;(web3Instance.eth.Contract as jest.Mock<unknown>).mockImplementation(() => ({
+        methods: {
+          allowance: jest.fn(() => ({
+            call: jest.fn(() => allowanceOnChain)
+          }))
+        }
+      }))
+
+      expect(await getAllowanceRequired({ quote, web3: web3Instance, erc20AllowanceAbi })).toEqual(
+        new BigNumber(900)
+      )
+    })
+  })
+
+  describe('getDeps', () => {
```

Comment:
kill this and the kung fu utility and just do `this.deps = deps` in constructor of `buildQuote`